src/blake2b_test.c

```diff
@@ -0,0 +1,88 @@
+#include "cbor.h"
+#include "test_utils.h"
+#include "hex_utils.h"
+#include "stream.h"
+#include <os.h>
+#include <string.h>
+
+void run_blake2b_test()
+{
+#define TESTCASE(input_, inputChunks_, chunksLength_, expected_) \
+	{ \
+	    stream_t si; \
+	    stream_init(&si); \
+	    stream_appendFromHexString(&si, input_); \
```

Comment:
The formatting looks good in my IDE

Reply:
Still - I presume this contains some weird mix of tabs and spaces

Reply:
Yep. (Note: astyle does not seem to fix indentation in macros). My vim shows it like this (>----- is tab)
```
#define TESTCASE(input_, inputChunks_, chunksLength_, expected_) \
>-------{ \
>-------    stream_t si; \
>-------    stream_init(&si); \
>-------    stream_appendFromHexString(&si, input_); \
>------->-------const uint8_t* inputBuffer = stream_head(&si); \
        stream_t se; \
>-------    stream_init(&se); \
>-------    stream_appendFromHexString(&se, expected_); \
>------->-------const uint8_t* expectedBuffer = stream_head(&se); \
        cx_blake2b_t hash; \
        cx_blake2b_init( \
                &hash, \
                64 * 8 /* output size in BITS */\
        ); \
        uint8_t output[64]; \
        uint8_t pos = 0; \
        for (unsigned i = 0; i < chunksLength_; i++) { \
            cx_hash( \
                &hash.header, \
                0, /* Do not output the hash, yet */ \
                inputBuffer + pos, \
                inputChunks_[i], /* Input length */ \
                NULL, 0 \
>-------        ); \
            pos += inputChunks_[i]; \
        } \
        cx_hash( \
>-------        &hash.header, \
>-------        CX_LAST, /* Output the hash */ \
>-------        inputBuffer, \
>-------        0, \
>-------        output, \
>-------        64 /* output length */ \
>-------    ); \
        EXPECT_EQ_BYTES(expectedBuffer, output, 64); \
    }
```